### PR TITLE
Add support for running a Flutter test in the debugger

### PR DESCRIPTION
**Pull request by skybrian**
The debug button is only activated when running all the tests in a file
(not a directory).

### CODE REVIEW

File: src/io/flutter/run/daemon/DaemonConsoleView.java [hunk 1]
```diff
@@ -37,11 +68,20 @@ public void print(@NotNull String text, @NotNull ConsoleViewContentType contentT
...
+    if (trimmed.startsWith("Observatory listening on http")) {
+      return;
+    }
+
+    if (trimmed.startsWith("Diagnostic server listening on http")) {
       return;
     }
 
-    if (trimmed.startsWith("Diagnostic server listening on http") && !trimmed.contains("\n")) {
+    if (trimmed.startsWith("You should first set appropriate breakpoints, then resume the test in the debugger.")) {
```
Comment (devoncarew) [inline src/io/flutter/run/daemon/DaemonConsoleView.java]: This comes from `pub test`? Perhaps a comment to that effect.

File: src/io/flutter/run/test/DebugTestRunner.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+          if (!outputType.equals(ProcessOutputTypes.STDOUT)) {
+            return;
+          }
+
+          final String text = event.getText().trim();
+          if (FlutterInitializer.isVerboseLogging()) {
+            LOG.info("[<-- " + text + "]");
+          }
+
+          if (!text.startsWith("[{") || !text.endsWith("}]")) {
```
Comment (devoncarew) [inline src/io/flutter/run/test/DebugTestRunner.java]: This is the format that `flutter test` returns?

File: third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceDebugProcessZ.java [hunk 3]
```diff
@@ -110,11 +110,6 @@ public boolean isRemoteDebug() {
     return remoteDebug;
   }
 
-  @NotNull
-  public ObservatoryConnector getConnector() {
```
Comment (devoncarew) [inline third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceDebugProcessZ.java]: Just to confirm, there are no callers, and we're not overriding (and changing behavior) from the parent class?

**Comment by devoncarew** [comment]
A few questions, but lgtm :)

File: third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceDebugProcessZ.java (hunk 3 above)
Comment (pq) [inline third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceDebugProcessZ.java]: At a quick glance, this looks dead to me.

Comment (skybrian) [inline third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceDebugProcessZ.java]: Yes, dead code.

File: src/io/flutter/run/daemon/DaemonConsoleView.java (hunk 1 above)
Comment (skybrian) [inline src/io/flutter/run/daemon/DaemonConsoleView.java]: It comes from flutter test. Added comment.

File: src/io/flutter/run/test/DebugTestRunner.java (hunk 2 above)
Comment (skybrian) [inline src/io/flutter/run/test/DebugTestRunner.java]: Yes, I used the same JSON format as daemon mode. But currently there's only one message, and it's only printed when using --machine and --start-paused.

**Pull request merged**